recipes/_helper/aws.py

```diff
@@ -54,25 +53,33 @@ def execute_query(
         Given query, database, output location, execute the query, 
         and print out temporary output location. check official 
         documentation for more detail.
-
-        Response Syntax:
-            {
-                'QueryExecutionId': 'string'
-            }
         """
 
-        queryStart = self.athena_client.start_query_execution(
+        queryStart = self.athena.start_query_execution(
             QueryString = query, 
             QueryExecutionContext = {'Database': database},
-            ResultConfiguration = {"OutputLocation": output_location}
+            ResultConfiguration = {"OutputLocation": f's3://{self.temporary_location}'}
         )
 
-        return queryStart
-
-    def move_file(self, source_location:str, target_location:str):
-        """
-        given source_location, move file from source_location to
-        target_location and remove the original file at source_location
-        """
-        return None
+        queryId = queryStart['QueryExecutionId']
+        queryLoc = f'{self.temporary_location}{queryId}.csv'
+        queryLoc_metadata = f'{self.temporary_location}{queryId}.csv.metadata'
+        
+        retry = True
```

Comment:
@mgraber the idea of this chunk is to have something that can wait till the query finishes, let me know if you have better ideas about how to do this. I feel uncomfortable with the while loop here.

When you execute a Athena query, it's basically a job in the que, and you get a query ID immediately,  but it doesn't mean the query is actually finished. So when we need to move the output, we would need to wait for the file to exist  first

Reply:
This is still a while loop, but at least it checks the query status directly:
https://www.ilkkapeltola.fi/2018/04/simple-way-to-query-amazon-athena-in.html